src/app/services/sources.ts

```diff
@@ -1,34 +1,57 @@
 import { DataSourceResponse } from '@/app/application/services/data-source/DataSourcePool'
 import type KumaApi from '@/services/kuma-api/KumaApi'
-import type {
-  PaginatedApiListResponse as CollectionResponse,
-} from '@/types/api.d'
-import type {
-  ServiceInsight,
-} from '@/types/index.d'
-
-type MeshParams = {
+import type { PaginatedApiListResponse as CollectionResponse } from '@/types/api.d'
+import type { ExternalService, ServiceInsight } from '@/types/index.d'
+
+type CollectionParams = {
   mesh: string
 }
+
+type DetailParams = CollectionParams & {
+  name: string
+}
+
 type PaginationParams = {
   size: number
   page: number
 }
+
 type Closeable = { close: () => void }
+
+export type ServiceInsightSource = DataSourceResponse<ServiceInsight>
 export type ServiceInsightCollection = CollectionResponse<ServiceInsight>
 export type ServiceInsightCollectionSource = DataSourceResponse<ServiceInsightCollection>
 
+export type ExternalServiceSource = DataSourceResponse<ExternalService>
+
 export const sources = (api: KumaApi) => {
   return {
-    '/:mesh/services': async (params: MeshParams & PaginationParams, source: Closeable) => {
+    '/:mesh/service-insights': async (params: CollectionParams & PaginationParams, source: Closeable) => {
       source.close()
+
+      const mesh = params.mesh
+      const size = params.size
       const offset = params.size * (params.page - 1)
-      return api.getAllServiceInsightsFromMesh({
-        mesh: params.mesh,
-      }, {
-        size: params.size,
-        offset,
-      })
+
+      return api.getAllServiceInsightsFromMesh({ mesh }, { size, offset })
+    },
+
+    '/:mesh/service-insights/:name': (params: DetailParams, source: Closeable) => {
+      source.close()
+
+      const mesh = params.mesh
+      const name = params.name
+
+      return api.getServiceInsight({ mesh, name })
+    },
+
+    '/:mesh/external-services/:name': (params: DetailParams, source: Closeable) => {
+      source.close()
+
+      const mesh = params.mesh
+      const name = params.name
```

Comment:
nitpick: could be `const { mesh, name } = { params.mesh, params.name }` right?

Reply:
Almost. It could use destructuring to look like this:

```js
const { mesh, name } = params
```

Let me change this when I work on the next pass over the detail views so I save myself the CI pipeline here.